src/Components/ColorForm/Theme.jsx

```diff
@@ -0,0 +1,199 @@
+  const handleDeleteTheme = () => {
+    const updatedThemes = themes.filter((theme) => theme.id !== selectedTheme);
+    setThemes(updatedThemes);
+
+    if (updatedThemes.length > 0) {
+      setSelectedTheme(updatedThemes[0].id);
+    } else {
+      setSelectedTheme(""); //
+    }
+
+    setIsConfirmingDelete(false);
+  };
+
+  const handleEditTheme = () => {
+    const updatedThemes = themes.map((theme) =>
+      theme.id === selectedTheme
+        ? {
+            ...theme,
+            name: newThemeName,
+            colors: theme.colors.map((color) => {
+              if (color.role === "primary main") {
+                return { ...color, hex: themeColors.primaryMain };
+              } else if (color.role === "primary dark") {
+                return { ...color, hex: themeColors.primaryDark };
+              } else if (color.role === "primary light") {
+                return { ...color, hex: themeColors.primaryLight };
+              } else {
+                return color;
+              }
+            }),
+          }
+        : theme
+    );
+
+    setThemes(updatedThemes);
+    setIsEditing(false);
+  };
+
+  const handleColorChange = (role, newColor) => {
+    setThemeColors((prevColors) => ({
+      ...prevColors,
+      [role]: newColor,
+    }));
+  };
+
+  return (
+    <div>
+      <select value={selectedTheme} onChange={handleThemeSelection}>
+        {themes.map((theme) => (
+          <option key={theme.id} value={theme.id}>
+            {theme.name}
+          </option>
+        ))}
+      </select>
+
+      <button className="btn_theme" onClick={handleAddTheme}>
+        Add Theme
+      </button>
+
+      {isConfirmingDelete ? (
+        <div>
+          <p>Are you sure you want to delete this theme?</p>{" "}
+          <button onClick={handleDeleteTheme}>Yes, Delete</button>{" "}
+          <button onClick={() => setIsConfirmingDelete(false)}>Cancel</button>{" "}
+        </div>
+      ) : (
+        <button
+          className="btn_theme"
+          onClick={() => setIsConfirmingDelete(true)}
+        >
+          Delete Theme
+        </button>
+      )}
+
+      {isEditing ? (
+        <div>
+          <input
+            type="text"
+            value={newThemeName}
+            onChange={(event) => setNewThemeName(event.target.value)}
+            placeholder="Enter new theme name"
+          />
+          <div>
+            <label>Primary Main: </label>
+
+            <input
+              type="color"
+              value={themeColors.primaryMain}
+              onChange={(event) =>
+                handleColorChange("primaryMain", event.target.value)
+              }
+            />
+          </div>
+          <div>
+            <label>Primary Dark: </label>
+
+            <input
+              type="color"
+              value={themeColors.primaryDark}
+              onChange={(event) =>
+                handleColorChange("primaryDark", event.target.value)
+              }
+            />
+          </div>
```

Comment:
unfortunately, your themes can now only work with three colors, and the names of these colors are fixed. This is not the desired behaviour. Each Theme needs to work as in the previous issues, plus that it has a name and can be switched. Therefore this PR needs to be worked over. I would suggest you start fresh over, since your code for switching and updating themes is based on this misconception.

Reply:
sh*t! 😪